components/eamxx/cime_config/namelist_defaults_scream.xml

```diff
@@ -348,6 +348,15 @@ be lost if SCREAM_HACK_XML is not enabled.
     <nc_nuceat_tend    >0.0</nc_nuceat_tend>
     <!-- SHOC internally inits TKE; just set to 0 here to signal it's not in the IC file -->
     <tke               >0.0</tke>
+    <!-- Rad heating and sfc fluxes are defined as "updated" so it will be written to restart files, so we   -->
```

Comment:
I'm adding a comment here just to ping @bartgol with a question.

Did we ever fix the code so that a variable can be initialized in the `init_impl` of a process and thus the AD won't try to load it from file.  I recall a long conversation about this when it came to SHOC, and we have this chunk of code: https://github.com/E3SM-Project/scream/blob/master/components/eamxx/src/physics/shoc/atmosphere_macrophysics.cpp#L278

Reply:
Yeah, this would be handy. I just had to add initialization for all of these changed fields to all of the input.yaml files in all of the tests that involve radiation. Not really a huge deal, but being able to tell the AD not to try to initialize fields that will be initialized by the individual processes would be nice. The way this is handled in EAM is that fields are assumed to be initialized in the individual parameterizations if they are not explicitly initialized via the initial condition file. Assuming is probably never the best idea, but maybe one solution would be to add some operation that tells the AD that a field has been initialized once it has been called. The AD could then check *after* all processes have ran init whether every field tagged as "updated" or "required"  has been initialized, either by IC, yaml constant initialization, or by calling this yet un-named initialization routine.

Reply:
By the way, I discovered what I think is kind of a weird bug in working on this. It seems that with a pg2 grid, the AD will NOT complain if a field tagged as "Updated" is not initialized, but with a np4 grid, I would get runtime fails without the extra initialization in the input.yaml for fields that were not in the IC. That is, this would fail:
```
ERS_D_Ln21.ne4_ne4.F2010-SCREAMv1.mappy_gnu9.scream-rad_frequency_2
```
but this would pass:
```
ERS_D_Ln21.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9.scream-rad_frequency_2
```

Reply:
Did you ever get to the bottom of this weird bug?  If not, can you make an issue so we don't lose sight of it?

Reply:
Yes, this is due to the currently slightly hacky impl of PG2 initialization. I have to fix this at some point...